File: cdk/lambda/lib/seedPrompt.js
/**
 * Returns the defaultPrompt if value is null, undefined, empty string,
 * or whitespace-only; otherwise returns value unchanged.
 *
 * @param {string|null|undefined} value - The value to check
 * @param {string} defaultPrompt - The fallback prompt to use
 * @returns {string} The original value or the default prompt
 */
function seedPrompt(value, defaultPrompt) {
  if (value === null || value === undefined || value.trim() === "") {
    return defaultPrompt;
  }
  return value;
}

module.exports = { seedPrompt };

File: cdk/text_generation/src/helpers/seed_prompt.py
"""Utility for seeding prompt values with defaults."""


def seed_prompt(value: str | None, default_prompt: str) -> str:
    """Return default_prompt if value is None, empty, or whitespace-only;
    otherwise return value unchanged.

    Args:
        value: The prompt value to check.
        default_prompt: The fallback prompt to use when value is blank.

    Returns:
        The original value or the default prompt.
    """
    if value is None or value.strip() == "":
        return default_prompt
    return value
